ztp/extra-manifests-builder/01-disk-encryption-pcr-rebind/luks-helpers.sh

```diff
@@ -0,0 +1,159 @@
+DEBUG="true"
+RESERVED_SLOT=31
+CLEVIS_CONFIG_RESERVED_SLOT="$RESERVED_SLOT: tpm2 '{\"hash\":\"sha256\",\"key\":\"ecc\"}'"
+TRUE=0
+FALSE=1
+NEWLINE=$'\n'
+
+#set -x
+
+# log function. Takes 2 arguments:
+# log level: debug or info
+# string to print
+log() {
+    case $1 in
+    "debug")
+        echo "DEBUG - $2" >&2
+        ;;
+    "info")
+        echo "INFO - $2" >&2
+        ;;
+    *)
+        # Code to execute when no patterns match
+        ;;
+    esac
+}
+
+# logs a string with a debug level
+logDebug() {
+    if ! [ -v DEBUG ] || ([ -v DEBUG ] && [ "$DEBUG" == "true" ]); then
+        log "debug" "$1"
+    fi
+}
+
+# logs a string with a info level
+logInfo() {
+    log "info" "$1"
+}
+
+# return $TRUE id the temporary reserved slot is configured with a key (to disable PCR protection), returns $FALSE otherwise
+isReservedSlotPresent() {
+    RESULT=$(clevis luks list -d $1 -s $RESERVED_SLOT || true)
+    if [ -n "$RESULT" ] && [ "$RESULT" == "$CLEVIS_CONFIG_RESERVED_SLOT" ]; then
+        logDebug "reserved slot $RESERVED_SLOT is present"
+        return $TRUE
+    fi
+    logDebug "reserved slot $RESERVED_SLOT is not present"
+    return $FALSE
+}
+
+# create a temporary key in the reserved slot to disable PCR protection
+addReservedSlot() {
+    logInfo "reservedSlotPresent=$1 device=$2 slot=$3 with PCR IDs=$4 and clevis config=$5"
+    if [ $1 == $TRUE ]; then
+        logInfo "reserve slot already present, no need to add again"
+        clevis luks list -d $2 || true
+        return
+    fi
+    logInfo "adding reserved slot on device=$device"
+    ANYPASS=$(openssl rand -base64 21)
+    echo -e "$ANYPASS\n" | clevis luks bind -s $RESERVED_SLOT -d $2 tpm2 '{}' || true
```

Comment:
I think the reserved slot content should be configurable (and I dont mind tpm2 only being the default)
but for tpm2+nbde we should at least be able to explain customers how to change the reserved slot config so it'll retain the nbde information.


an even nicer way can be if you copy that from the original slot but that can be tricky.


the other option, is maybe ditch tpm2+nbde
might make it all cleaner
then we can revisit that and how to properly do it in the future
@lack WDYT?

Reply:
the script already engage the reserved slot only if there are PCRs.. (so there is tpm2..)

Reply:
removed the nbde+tpm2 scenario in this PR